packages/dashboards/src/lib/components/timeseries-widget/chart-presets/xy-chart/xy-chart.component.ts

```diff
@@ -85,6 +90,28 @@ export abstract class XYChartComponent extends TimeseriesChartComponent
         }
     }
 
+    public onInteraction(legendSeries: IChartAssistSeries<IAccessors>, event: MouseEvent) {
+        let interactive = this.isSeriesInteractive(legendSeries);
+        if (!interactive) {
+            console.log("📘 xy-chart.component: 96# -> legendSeries:", legendSeries);
+            this.eventBus.getStream(INTERACTION).next({
+                payload: {
+                    data: legendSeries,
+                    interactionType: TimeseriesInteractionType.Series,
+                },
+            });
+            return;
+        }
+        const target = event.target as HTMLElement;
+        const secondaryDescription = target.classList.contains("legend-description-secondary");
+
+        if (legendSeries.link && !secondaryDescription) {
+            this.eventBus.getStream(INTERACTION).next({ payload: { data: legendSeries } });
+        } else if (secondaryDescription) {
+            this.chartAssist.toggleSeries(legendSeries.id, this.chartAssist.isSeriesHidden(legendSeries.id));
+        }
+    }
```

Comment:
Would it be possible to separate this logic out into two separate click handlers? One for `onPrimaryDescClick` and one for `onSecondaryDescClick`. That may help to simplify the logic here, as you wouldn't have to query the event for the target class list.